src/test/java/itest/GraphQLIT.java

```diff
@@ -996,6 +997,1144 @@ public void testQueryforFilteredEnvironmentNodesByNames() throws Exception {
+                                                ar.result().bodyAsString(),
+                                                StartRecordingMutationResponse.class));
+                                latch.countDown();
+                            }
+                        });
+
+        latch.await(30, TimeUnit.SECONDS);
+
+        // Ensure ActiveRecording created  is STOPPED
+        JsonObject notification2 = f2.get(5, TimeUnit.SECONDS);
+        JsonObject notificationRecording2 =
+                notification2.getJsonObject("message").getJsonObject("recording");
+
+        Assertions.assertEquals("test", notificationRecording2.getString("name"));
+        Assertions.assertEquals("STOPPED", notificationRecording2.getString("state"));
+
+        // Restart the recording
+        CompletableFuture<StartRecordingMutationResponse> resp3 = new CompletableFuture<>();
+        query.put(
+                "query",
+                "query { targetNodes(filter: { annotations: \"PORT == 9093\" }) {"
+                    + " doStartRecording(recording: { name: \"test\", duration: 30, template:"
+                    + " \"Profiling\", templateType: \"TARGET\", replace:ALWAYS }) { name state}}"
+                    + " }");
+
+        Future<JsonObject> f3 =
+                worker.submit(
+                        () -> {
+                            try {
+                                return expectNotification(
+                                                "ActiveRecordingCreated", 15, TimeUnit.SECONDS)
+                                        .get();
+                            } catch (Exception e) {
+                                throw new RuntimeException(e);
+                            } finally {
+                                latch.countDown();
+                            }
+                        });
+
+        Thread.sleep(5000);
+        webClient
+                .post("/api/v2.2/graphql")
+                .sendJson(
+                        query,
+                        ar -> {
+                            if (assertRequestStatus(ar, resp3)) {
+                                resp3.complete(
+                                        gson.fromJson(
+                                                ar.result().bodyAsString(),
+                                                StartRecordingMutationResponse.class));
+                                latch.countDown();
+                            }
+                        });
+
+        latch.await(30, TimeUnit.SECONDS);
+
+        // Ensure ActiveR ecording is Recreated
+        JsonObject notification3 = f3.get(5, TimeUnit.SECONDS);
+        JsonObject notificationRecording3 =
+                notification3.getJsonObject("message").getJsonObject("recording");
+
+        Assertions.assertEquals("test", notificationRecording3.getString("name"));
+        Assertions.assertEquals("RUNNING", notificationRecording3.getString("state"));
+
+        // Delete the Recording
```

Comment:
To be extra-safe, everything above should be wrapped in a `try { }` and this deletion operation should be in a `} finally { }`, so that the recording gets cleaned up even if the preceding assertions fail